pkg/cli/cmds/root.go

```diff
@@ -245,7 +246,10 @@ func validateCISReqs(role CLIRole) error {
 	// etcd user only needs to exist on servers
 	if role == Server {
 		if _, err := user.Lookup("etcd"); err != nil {
-			ce = append(ce, fmt.Errorf("missing required %w", err))
+			ce = append(ce, errors.Wrap(err, "missing required"))
```

Comment:
I'll defer to @ShylajaDevadiga but I believe we also need to check that the `Gid` of the user returned by the `user.Lookup` call is the same as the `Gid` of the group from `user.LookupGroup`. If we ensure that here, the bits at https://github.com/rancher/rke2/blob/6f2632aaae172dda758813c1dc3c8928ba36682f/pkg/podexecutor/staticpod.go#L521 should fix the directory ownership for us.

Reply:
There's a chance that the uid and gid don't match depending on how the operator has created the groups. I just did a test where the uid and gid are different and user and groups are assigned correctly in the pod manifest as well as the applied ownership of the db directory.

Reply:
Ah, OK. The policy just says we need both an etcd user and group, and that the directory must be owned by them... but it doesn't have any requirement around the user's primary gid or membership in the etcd group. Fair enough.

Reply:
uid and gid need not be the same while creating the user, we only need etcd user and etcd group to exist
etcd:x:990:987:etcd user:/home/etcd:/sbin/nologin